src/relocate.rs

```diff
@@ -76,7 +76,7 @@ macro_rules! debug_assert_eq {
 /// In practice, things work if we don't make any calls to functions outside
 /// of this crate, not counting functions directly implemented by the compiler.
 /// So we can do eg. `x == null()` but we can't do `x.is_null()`, because
-/// `null` is directly implemented by the compiler, while `is_null` is not.
+/// `null` is directly implemented by the compiler, while `is_null` is not 🙃.
```

Comment:
This patch cause segfaults:

```patch
diff --git a/src/relocate.rs b/src/relocate.rs
index 7eddd09..29b5e2b 100644
--- a/src/relocate.rs
+++ b/src/relocate.rs
@@ -375,8 +375,7 @@ unsafe fn compute_auxp(envp: *mut *mut u8) -> *const Elf_auxv_t {
     // Locate the AUX records we need. We don't use rustix to do this because
     // that would involve calling a function in another crate.
     let mut auxp = envp;
-    // Don't use `is_null` because that's a call.
-    while (*auxp) != null_mut() {
+    while !(*auxp).is_null() {
         auxp = auxp.add(1);
     }
     auxp.add(1).cast()
```